lib/hd/keyorigin.js

```diff
@@ -0,0 +1,173 @@
+/*!
+ * keyorigin.js - hd key path object for bcoin
+ * Copyright (c) 2023, the bcoin developers (MIT License).
+ * https://github.com/bcoin-org/bcoin
+ */
+
+'use strict';
+
+const assert = require('assert');
+const common = require('./common');
+const bio = require('bufio');
+const {inspectSymbol} = require('../utils');
+/**
+ * Helper class to represent hd key path for arbitrary wallets.
+ * @property {Number} fingerPrint - master key fingerprint (uint32)
+ * @property {Array} path - bip32 derivation path in uint32 array
+ */
+class KeyOriginInfo {
+  constructor(options) {
+    this.fingerPrint = -1;
+    this.path = [];
+    if (options) {
+      this.fromOptions(options);
+    }
+  }
+
+  fromOptions(options) {
+    assert(options, 'requires options');
+
+    if (options.fingerPrint !== undefined) {
+      assert(
+        options.fingerPrint >>> 0 === options.fingerPrint &&
+          options.fingerPrint <= 0xffffffff,
```

Comment:
I think this second check is redundant. That's the nice thing about the `>>>` operator

```
> 0xffffffff >>> 0 === 0xffffffff
true
> (0xffffffff + 1) >>> 0 === (0xffffffff + 1)
false
> (0xffffffff - 1) >>> 0 === (0xffffffff - 1)
true
```